**Data source**: http://datos.energia.gob.ar/dataset/produccion-de-petroleo-y-gas-por-pozo

*“Monthly production breakdown by well, field, concession, and province. Oil [m³], gas [thousand m³], and water [m³].”*

In [2]:
import pandas as pd

# Cargar el archivo CSV en un DataFrame
df = pd.read_csv('../data/raw/produccin-de-pozos-de-gas-y-petrleo-2026.csv')

In [4]:
df.shape

(328917, 38)

In [3]:
df.head()

,idempresa,anio,mes,idpozo,prod_pet,prod_gas,prod_agua,iny_agua,iny_gas,iny_co2,...,idareayacimiento,areayacimiento,cuenca,provincia,tipo_de_recurso,proyecto,clasificacion,subclasificacion,sub_tipo_recurso,fecha_data
0,YPF,2026,1,115810,2.46,34.97,0.00,0.0,0.0,0.0,...,LDL,LOMA LA LATA,NEUQUINA,Neuquén,CONVENCIONAL,Sin Proyecto,EXPLOTACION,DESARROLLO,NaN,2026-01-31
1,YPF,2026,1,740,0.00,0.00,0.00,0.0,0.0,0.0,...,PPCO,PUESTO POZO CERCADO OCCIDENTAL,CUYANA,Mendoza,CONVENCIONAL,Sin Proyecto,NaN,NaN,NaN,2026-01-31
2,YPF,2026,1,128473,0.53,0.19,75.93,0.0,0.0,0.0,...,HER,PUESTO HERNANDEZ,NEUQUINA,Neuquén,CONVENCIONAL,Sin Proyecto,EXPLOTACION,DESARROLLO,NaN,2026-01-31
3,YPF,2026,1,115812,12.06,253.00,0.00,0.0,0.0,0.0,...,LDL,LOMA LA LATA,NEUQUINA,Neuquén,CONVENCIONAL,Sin Proyecto,EXPLOTACION,DESARROLLO,NaN,2026-01-31
4,YPF,2026,1,112655,0.00,0.00,0.00,0.0,0.0,0.0,...,SPI,SEÑAL PICADA,NEUQUINA,Rio Negro,CONVENCIONAL,Sin Proyecto,EXPLOTACION,AVANZADA,NaN,2026-01-31


In [6]:
df.columns

Index(['idempresa', 'anio', 'mes', 'idpozo', 'prod_pet', 'prod_gas',
       'prod_agua', 'iny_agua', 'iny_gas', 'iny_co2', 'iny_otro', 'tef',
       'vida_util', 'tipoextraccion', 'tipoestado', 'tipopozo',
       'observaciones', 'fechaingreso', 'rectificado', 'habilitado',
       'idusuario', 'empresa', 'sigla', 'formprod', 'profundidad', 'formacion',
       'idareapermisoconcesion', 'areapermisoconcesion', 'idareayacimiento',
       'areayacimiento', 'cuenca', 'provincia', 'tipo_de_recurso', 'proyecto',
       'clasificacion', 'subclasificacion', 'sub_tipo_recurso', 'fecha_data'],
      dtype='str')

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 328917 entries, 0 to 328916
Data columns (total 38 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   idempresa               328917 non-null  str    
 1   anio                    328917 non-null  int64  
 2   mes                     328917 non-null  int64  
 3   idpozo                  328917 non-null  int64  
 4   prod_pet                328917 non-null  float64
 5   prod_gas                328917 non-null  float64
 6   prod_agua               328917 non-null  float64
 7   iny_agua                328917 non-null  float64
 8   iny_gas                 328917 non-null  float64
 9   iny_co2                 328917 non-null  float64
 10  iny_otro                328917 non-null  float64
 11  tef                     328917 non-null  float64
 12  vida_util               28877 non-null   float64
 13  tipoextraccion          328899 non-null  str    
 14  tipoestado              328899 

idempresa (string): Identifier of the company operating the well

anio (integer): Reported Year

mes (integer): Reported Month

idpozo (integer): Well identifier by productive formation

prod_pet (number): Oil production (in m3)

prod_gas (number): Gas production (in thousands of m3)

prod_agua (number): Water production (in m3)

iny_agua (number): Water injection (in m3)

iny_gas (number): Gas injection (in thousands of m3)

iny_co2 (number): CO2 injection (in thousands of m3)

iny_otro(number): Injection of other products (in m3)

tef (number): Effective well operating time

vida_util (number): Estimated useful life of the well

tipoextraccion (string): Type of fluid extraction from the well

tipoestado (string): Well status

tipopozo (string): Well operating characteristics

observaciones(string): Observations

fechaingreso (date-time): Data upload date

rectificado (boolean): The data was corrected by the operator

habilitado (boolean): Registration enabled for publication

idusuario (integer): User identifier for data upload

empresa (string): Name of the operating company

sigla (string): Unique identifier of the wellhead

formprod (string): Producing formation identifier

profundidad (number): Well depth

formacion (string): Producing formation

idareapermisoconcesion (string): Area, permit, or concession identifier

areapermisoconcesion (string): Area, permit, or concession name

idareayacimiento (string): Field identifier

areayacimiento (string): Field name

cuenca (string): Sedimentary basin name

provincia (string): Name of the province

tipo_de_recurso (string): Resource type

proyecto (string): Name of the project associated with the well

clasificacion (string): Well classification

subclasificacion (string): Well subclassification

sub_tipo_recurso (string): Type of hydrocarbon resource



**Source of variable descriptions (in Spanish):**
*https://github.com/datosenergia/produccion-de-petroleo-y-gas-por-pozo*

## Variable Categories

### Production Metrics
- prod_pet
- prod_gas
- prod_agua

### Injection Metrics
- iny_agua
- iny_gas
- iny_co2
- iny_otro

### Temporal Information
- anio
- mes
- fechaingreso

### Geographic Information
- provincia
- cuenca
- areayacimiento

### Well Characteristics
- profundidad
- vida_util
- tipoextraccion
- tipopozo

### Resource Classification
- tipo_de_recurso
- clasificacion
- subclasificacion
- sub_tipo_recurso

### Administrative Metadata
- habilitado
- rectificado
- idusuario

In [8]:
df.describe()

,anio,mes,idpozo,prod_pet,prod_gas,prod_agua,iny_agua,iny_gas,iny_co2,iny_otro,tef,vida_util,idusuario,profundidad
count,328917.0,328917.000000,328917.000000,3.289170e+05,328917.000000,328917.000000,328917.000000,3.289170e+05,328917.0,328917.000000,328917.000000,28877.0,328917.000000,328917.000000
mean,2026.0,2.496633,103935.660106,5.119101e+01,50.625866,318.983748,319.130717,1.839471e+01,0.0,1.537962,9.318320,0.0,409.918153,1732.082317
std,0.0,1.130897,42834.813771,3.471300e+02,820.200689,1111.437236,1921.890611,6.666872e+03,0.0,221.494988,16.979259,0.0,60.218695,1761.401901
min,2026.0,1.000000,212.000000,-5.000000e-16,0.000000,0.000000,0.000000,0.000000e+00,0.0,0.000000,0.000000,0.0,3.000000,0.000000
25%,2026.0,1.000000,71827.000000,0.000000e+00,0.000000,0.000000,0.000000,0.000000e+00,0.0,0.000000,0.000000,0.0,353.000000,1055.000000
50%,2026.0,2.000000,111607.000000,0.000000e+00,0.000000,0.000000,0.000000,0.000000e+00,0.0,0.000000,0.000000,0.0,420.000000,1604.000000
75%,2026.0,3.000000,134284.000000,3.990000e+00,0.000000,6.410000,0.000000,0.000000e+00,0.0,0.000000,27.960000,0.0,473.000000,2362.000000
max,2026.0,5.000000,167247.000000,1.457872e+04,104141.900000,28671.300000,178089.891000,2.791888e+06,0.0,42971.800000,719.991944,0.0,497.000000,378939.000000


In [12]:
for col in ["habilitado", "rectificado"]:
    print(df[col].value_counts(dropna=False))

habilitado
t    328917
Name: count, dtype: int64
rectificado
f    328917
Name: count, dtype: int64


In [14]:
df["idpozo"].nunique()

83083

In [15]:
len(df)

328917

In [16]:
df.groupby("idpozo").size().describe()

count    83083.000000
mean         3.958897
std          0.348710
min          1.000000
25%          4.000000
50%          4.000000
75%          4.000000
max          5.000000
dtype: float64

In [17]:
df.duplicated(
    subset=["idpozo", "anio", "mes"]
).sum()

np.int64(0)

In [9]:
df.isnull().sum()

idempresa                      0
anio                           0
mes                            0
idpozo                         0
prod_pet                       0
prod_gas                       0
prod_agua                      0
iny_agua                       0
iny_gas                        0
iny_co2                        0
iny_otro                       0
tef                            0
vida_util                 300040
tipoextraccion                18
tipoestado                    18
tipopozo                      18
observaciones             279338
fechaingreso                   0
rectificado                    0
habilitado                     0
idusuario                      0
empresa                        0
sigla                          0
formprod                   10716
profundidad                    0
formacion                  11142
idareapermisoconcesion         0
areapermisoconcesion           0
idareayacimiento               0
areayacimiento                 0
cuenca    

In [10]:
df.duplicated().sum()

np.int64(0)

### Dataset Size

- 328,917 records
- 83,083 unique wells

### Granularity

- One record per well per month

### Data Quality

- No duplicated records for (idpozo, anio, mes)

### Administrative Fields

- habilitado = True for all records
- rectificado = False for all records
